hooks/base16-delta.sh

```diff
@@ -37,7 +37,7 @@ b_hex_value=${current_bg_color:4:2}
 # 1.) The final result is_light_theme=1 or is_light_theme=0
 # 2.) The relevant values & HSP calculations that is in Bash comments,
 #     and will be added in the header section of the generated gitconfig.
-bc_output=$(bc -g <<-HSP_BC
+bc_output=$(bc <<-HSP_BC
```

Comment:
I think the missing support for the `-g` flag in some `bc` versions is the root of the problem. 

All it does is ensure that changes to `ibase|obase` are managed as a stack, otherwise they are treated as global. With the absence of this option, the call to `ibase=16` in `hex2dec` as interpreted as `16 base 16` of the expected `16 base 10`.

This works on macOS's `bc`:

```
❯ bc --version
bc 4.0.2
Copyright (c) 2018-2021 Gavin D. Howard and contributors
Report bugs at: https://git.yzena.com/gavin/bc
```

Since we apparently can't rely on that, we should probably update `hex2dec` to set `ibase|obase` globally, properly:

```bc
scale=3
# Stash the base values in decimal form before we switch to interpreting all values as hexadecimal.
hexbase=16
decbase=10
obase=decbase
ibase=hexbase
define hex2dec(x) {
   ibase=hexbase
   obase=decbase
   return(x)
}

# etc.
```

Reply:
Actually I think `hex2dec` is unnecessary. What does this give you in your version of `bc`?

```
bc <<-HSP_BC
scale=3
obase=10
ibase=16

r=$r_hex_value
g=$g_hex_value
b=$b_hex_value
hsp=sqrt((.4C8 * ${r_hex_value} ^ 2) + (0.964 * ${g_hex_value} ^ 2) + (.1D2 * ${b_hex_value} ^ 2))

is_light_theme=(hsp>7F.8)

# gitconfig headers (Bash comments)
print "# hex            = #$current_bg_color\n"
print "# r              = ", r, "\n"
print "# g              = ", g, "\n"
print "# b              = ", b, "\n"
print "# hsp            = sqrt(.299r^2 + .587g^2 + .114b^2)\n"
print "# hsp            = ", hsp, "\n"
print "# is_light_theme = hsp > 127.5\n"
print "# is_light_theme = ", is_light_theme, "\n"

# Variable assignment to Bash variable is_light_theme
print "is_light_theme=", is_light_theme
HSP_BC
```

Reply:
This is what I get with `bc 4.0.2`:

```
❯ base16_ayu-dark
# hex            = #0F1419
# r              = 15
# g              = 20
# b              = 25
# hsp            = sqrt(.299r^2 + .587g^2 + .114b^2)
# hsp            = 19.289
# is_light_theme = hsp > 127.5
# is_light_theme = 0
is_light_theme=0
                                                                                      14:17:38
❯ base16_ayu-light
# hex            = #FAFAFA
# r              = 250
# g              = 250
# b              = 250
# hsp            = sqrt(.299r^2 + .587g^2 + .114b^2)
# hsp            = 249.624
# is_light_theme = hsp > 127.5
# is_light_theme = 1
is_light_theme=1
```

Reply:
Nice, I've updated with your suggested edits and it works for me.

![image](https://github.com/tinted-theming/base16-shell/assets/602472/ccae5b4b-6f50-4fce-b717-13343e35a040)

![image](https://github.com/tinted-theming/base16-shell/assets/602472/25f3b9c2-fd88-479a-93bd-9230ef468a3e)

Thanks for looking into it!

Reply:
🙌